client/src/components/RuleBuilder/RuleComponent.vue

```diff
@@ -40,7 +37,7 @@ export default {
             this.$emit("update:displayRuleType", null);
         },
         okay() {
-            this.builder.handleRuleSave(this.ruleType);
+            this.$emit("save-rule", this.ruleType);
```

Comment:
The events casing is inconsistent, which seems to lead to it not being handled.

```suggestion
            this.$emit("saveRule", this.ruleType);
```